test/e2e-olm/elasticsearch_test.go

```diff
@@ -1,159 +1,302 @@
-			elasticsearch.ElasticsearchRoleMaster,
-		},
-		NodeCount: int32(1),
-		Storage: elasticsearch.ElasticsearchStorageSpec{
-			StorageClassName: &storageClassName,
-			Size:             &storageClassSize,
-		},
-		GenUUID: &nonDataUUID,
+	ctx.Cleanup()
+	e2eutil.WaitForDeletion(t, f.Client.Client, cr, cleanupRetryInterval, cleanupTimeout)
+	t.Log("Finished successfully")
+}
+
+func multipleNodesTest(t *testing.T) {
+	f := test.Global
+
+	ctx := test.NewContext(t)
+	namespace, err := ctx.GetWatchNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Found namespace: %v", namespace)
+
+	esUUID := utils.GenerateUUID()
+	t.Logf("Using UUID for elasticsearch CR: %v", esUUID)
+
+	if err = createElasticsearchSecret(t, f, ctx, esUUID); err != nil {
+		t.Fatal(err)
+	}
+
+	dataUUID := utils.GenerateUUID()
+	t.Logf("Using GenUUID for data nodes: %v", dataUUID)
+
+	// Create CR with two nodes sharing client, data and master roles
+	cr, err := createElasticsearchCR(t, f, ctx, esUUID, dataUUID, 2)
+	if err != nil {
+		t.Fatalf("could not create exampleElasticsearch: %v", err)
 	}
 
-	exampleElasticsearch, err := createElasticsearchCR(t, f, ctx, dataUUID)
+	dplName := fmt.Sprintf("elasticsearch-%v-cdm-%v-1", esUUID, dataUUID)
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, dplName, 1, retryInterval, timeout)
 	if err != nil {
-		return fmt.Errorf("could not create exampleElasticsearch: %v", err)
+		t.Fatalf("timed out waiting for first node deployment %v: %v", dplName, err)
 	}
 
-	deploymentName := fmt.Sprintf("elasticsearch-cdm-%v-1", dataUUID)
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, deploymentName, 1, retryInterval, timeout)
+	dplName = fmt.Sprintf("elasticsearch-%v-cdm-%v-2", esUUID, dataUUID)
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, dplName, 1, retryInterval, timeout)
 	if err != nil {
-		return fmt.Errorf("timed out waiting for initial Deployment %v: %v", deploymentName, err)
+		t.Fatalf("timed out waiting for second node deployment %v: %v", dplName, err)
 	}
-	t.Log("Created initial deployment")
 
-	// Scale up current node
-	// then look for elasticsearch-cdm-0-2 and prior node
-	t.Log("Scaling up the current node...")
-	exampleName := types.NamespacedName{Name: elasticsearchCRName, Namespace: namespace}
-	if err = f.Client.Get(goctx.TODO(), exampleName, exampleElasticsearch); err != nil {
-		return fmt.Errorf("failed to get exampleElasticsearch: %v", err)
+	ctx.Cleanup()
+	e2eutil.WaitForDeletion(t, f.Client.Client, cr, cleanupRetryInterval, cleanupTimeout)
+	t.Log("Finished successfully")
+}
+
+func scaleUpNodesTest(t *testing.T) {
+	f := test.Global
+
+	ctx := test.NewContext(t)
+	namespace, err := ctx.GetWatchNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Found namespace: %v", namespace)
+
+	esUUID := utils.GenerateUUID()
+	t.Logf("Using UUID for elasticsearch CR: %v", esUUID)
+
+	if err = createElasticsearchSecret(t, f, ctx, esUUID); err != nil {
+		t.Fatal(err)
```

Comment:
this seems to all be duplicated from test to test, can we move it into a `setup()` function of some sort?